test/y2storage/initial_guided_proposal_multidisk_first_test.rb

```diff
@@ -0,0 +1,339 @@
+          end
+
+          it "proposes only required volumes" do
+            proposal.propose
+
+            expect(proposed_vg_names).to_not include("vg-pgsql")
+          end
+        end
+      end
+
+      context "and allocate_volume_mode set to :device" do
+        let(:allocate_mode) { :device }
+
+        context "using LVM" do
+          let(:lvm) { true }
+
+          context "with separated volume groups" do
+            let(:separate_vgs) { true }
+
+            include_examples "make a proposal using as much devices as possible"
+          end
+
+          context "without separated volume groups" do
+            let(:separate_vgs) { false }
+
+            # NOTE: even setting the allocate_device_mode to :device, the
+            # proposal will use only one device when separated volumes are not
+            # required.
+            context "and some devices has enough space to hold the proposal" do
+              let(:sdc_size) { 500.GiB }
+
+              it "makes the proposal using the biggest one" do
+                proposal.propose
+
+                expect(used_devices).to contain_exactly("/dev/sdc")
+              end
+            end
+
+            context "and none device has enough space to hold the proposal" do
+              let(:sda_size) { 5.GiB }
+              let(:sdb_size) { 5.GiB }
+              let(:sdc_size) { 5.GiB }
+              let(:sdd_size) { 5.GiB }
+
+              it "raises a NoDiskSpaceError (although all devices together could hold it)" do
+                expect { proposal.propose }.to raise_exception(Y2Storage::NoDiskSpaceError)
+              end
+            end
+          end
+        end
+
+        context "not using LVM" do
+          let(:lvm) { false }
+
+          context "with separated volume groups" do
+            let(:separate_vgs) { true }
+
+            include_examples "make a proposal using as much devices as possible"
+          end
+
+          context "without separated volume groups" do
+            let(:separate_vgs) { false }
+
+            it "spreads the proposal as much as possible" do
+              proposal.propose
+
+              expect(used_devices).to contain_exactly("/dev/sda", "/dev/sdb", "/dev/sdd")
+            end
+          end
+        end
+      end
+
+      # NOTE: this scenario, multidisk_first + allocate_volume_mode :auto, is
+      # a kind of "fallback" to the behavior of the initial proposal before
+      # changes introduced in https://github.com/yast/yast-storage-ng/pull/783
+      context "with allocate_volume_mode set to :auto" do
+        let(:allocate_mode) { :auto }
+
+        context "using LVM" do
+          let(:lvm) { true }
+
+          context "with separated volume groups" do
+            let(:separate_vgs) { true }
+
+            it "spreads the proposal as much as possible" do
+              proposal.propose
+
+              expect(used_devices).to contain_exactly("/dev/sda", "/dev/sdb", "/dev/sdd")
```

Comment:
Umm, could you try it with a very very big sda? I would expect it only uses the necessary devices.